polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java

```diff
@@ -91,6 +91,7 @@ public void setPolarisAuthenticator(
   }
 
   public RealmContextResolver getRealmContextResolver() {
+    realmContextResolver.setDefaultRealm(this.defaultRealm);
     return realmContextResolver;
```

Comment:
I'm confused by this suggestion; the linked issue is about users not being able to pass an explicit string default via the config. Are you suggesting that we remove the config?

Reply:
ah, I wasn't clear in the first place. I think we can combine these two configs to one. 
```
realmContextResolver:
  type: default

defaultRealms:
  - default-realm
```
To 
```
realmContextResolver:
  type: default
  defaultRealms: default-realm
 ```

With that, we don't have to keep two fields in the config class, and I think it is a nice way to manage the lifecycle of `realmContextResolver`. It could be immutable object without any setting methods.

Reply:
If we made the above change, I think we don't have to make the interface change in `RealmContextResolver`. We just need to add a constructor in the class `DefaultContextResolver` like this
```
public DefaultContextResolver( @JsonProperty("defaultRealms") final String defaultRealm) {
    this.defaultRealm = defaultRealm;
}
```

Reply:
Removing the default constructor is still a breaking change, so is this materially better?

Reply:
Also, to be clear, we are not just changing how this works for `DefaultContextResolver` but for all `RealmContextResolver` implementations

Reply:
The `defaultRealm` is just the default realm used by `RealmContextResolver` when the realm cannot be parsed from the request

Reply:
Synced with @eric-maynard offline, we agreed on pushing this break change to another PR.